lib/sles4sap/sap_deployment_automation_framework/networking.pm

```diff
@@ -0,0 +1,389 @@
+
+=back
+
+Returns names of lease blob files within 'network-spaces' container which are older than retention time.
+Test will search network spaces which were reserved more than B<$check_older_than_sec> seconds ago.
+Default 7h should be plenty for not triggering race condition between network assignment and actual infrastructure creation.
+
+=cut
+
+sub list_expired_files {
+    my ($check_older_than_sec) = @_;
+    $check_older_than_sec //= 7 * 3600;    # 7 hours default
+    my @free_lease_files;
+    # this creates list of network file names and last modification time
+    my $all_files = az_storage_blob_list(
+        container_name => 'network-spaces',
+        storage_account_name => get_required_var('SDAF_TFSTATE_STORAGE_ACCOUNT'),
+        query => "[].{network:name , last_modified:properties.lastModified}"
+    );
+
+    foreach (@$all_files) {
+        # for format explanation check 'man strftime' - az cli returns date in ISO 8601 format
+        # Time::Piece does not recognize az cli timezone format `+02:00`, only `+0200`
+        # therefore we have to do some colonoscopy and remove `:` from az cli output
+        $_->{last_modified} =~ s/(?<=\+\d\d):(?=\d\d$)//;
+        # Avoid using `%F` instead of `%F` as it is not supported in older perl versions
+        my $time = Time::Piece->strptime($_->{last_modified}, '%Y-%m-%dT%H:%M:%S%z')->epoch();
+        push(@free_lease_files, $_->{network}) if $time < time() - $check_older_than_sec;
+    }
+    return @free_lease_files;
+}
+
+=head2 list_network_lease_files
+
+    list_network_lease_files();
+
+Returns names of all lease blob files within 'network-spaces' container.
+
+=cut
+
+sub list_network_lease_files {
+    return az_storage_blob_list(
+        container_name => 'network-spaces',
+        storage_account_name => get_required_var('SDAF_TFSTATE_STORAGE_ACCOUNT'),
+        query => "[].name"
+    );
+}
+
+=head2 acquire_network_file_lease
+
+    acquire_network_file_lease(network_lease_file=>'192.168.1.0' [, storage_account=>'some account']);
+
+=over 2
+
+=item B<storage_account>: Storage account containing lease file. Default: 'SDAF_TFSTATE_STORAGE_ACCOUNT'
+
+=item B<network_lease_file>: Name of the lease file
+
+=back
+
+Acquire network lease for a blob.
+
+=cut
+
+sub acquire_network_file_lease {
+    my (%args) = @_;
+    $args{storage_account} //= get_required_var('SDAF_TFSTATE_STORAGE_ACCOUNT');
+    foreach ('network_lease_file', 'storage_account') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    my $lease_id;
+    my %arguments = (
+        container_name => 'network-spaces',
+        storage_account_name => $args{storage_account},
+        blob_name => $args{network_lease_file},
+        lease_duration => 60
+    );
+
+    # First acquire a file lease to gain exclusive file access - prevents other tests modifying it.
+    $lease_id = az_storage_blob_lease_acquire(%arguments);
```

Comment:
Add it here as:
```
my $lease_id = az_storage_blob_lease_acquire(%arguments);
```

Reply:
Yeah that makes sense. It might have been a remnant of an older code structure. 
Thanks. Fixed.